src/TLuaInterpreterMudletObjects.cpp

```diff
@@ -1128,7 +1128,8 @@ int TLuaInterpreter::permRegexTrigger(lua_State* L)
     }
 
     const QString script{lua_tostring(L, 4)};
-    auto [triggerId, message] = pLuaInterpreter->startPermRegexTrigger(name, parent, regList, script);
+    bool multiline = lua_isboolean(L, 5) ? lua_toboolean(L, 5) : false;
```

Comment:
Given that the fifth argument is optional - does Lua count a missing argument as a "falsey" value - i.e. will `lua_isboolean(...)` on a nil produce an error or coerce it to `false`. If it doesn't and this is a valid means to detect its value if provided - then don't we have to default to a `true` if there is more than one item in the table of "items" for a "root" level trigger so that we maintain our ***backward-compatiblity*** (behaviour) promise?

Also, don't we have a `getVerifiedBool(...)` method to (error) handle some of this (though it doesn't IIRC handle the presence/absence of an argument so that has to be gated behind a `lua_gettop(L)` test?

Reply:
`nil` in Lua is false, so no backwards compatibility is broken. This can be tested by trying the build available here :wink: 

Adjusted to use `getVerifiedBool` instead.

Reply:
If the user does not supply that fifth "multi-line" option (as previous scripts would not do) then if they supply more than one "item" then prior behaviour would make it a multi-line trigger rather than a multi-item one. So doesn't the option have to default to `true` (multi-line AND trigger) not `false` (multi-item OR trigger) if it is not given?